subsys/pcd/src/pcd.c

```diff
@@ -75,6 +80,24 @@ const void *pcd_cmd_data_ptr_get(void)
 }
 
 #ifdef CONFIG_PCD_NET
+int pcd_find_fw_version(void)
+{
+	const struct fw_info *firmware_info;
+
+	if (cmd->magic != PCD_CMD_MAGIC_READ_VERSION) {
+		return -EFAULT;
+	}
+
+	firmware_info = fw_info_find(PM_APP_ADDRESS);
+
+	if (firmware_info != NULL) {
+		memcpy((void *)cmd->data, &firmware_info->version, cmd->len);
```

Comment:
Are we sure that the size of data to copy is taken from the destination buffer?

Reply:
Pretty much yes as the command which sets `PCD_CMD_MAGIC_READ_VERSION` also sets `len` and `data` but if we want to ensure that the API can't be misused I can add another check.